[fmilib] Build error

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Linux CentOS7
- Compiler: gcc (GCC) 8.3.1 20190311 (Red Hat 8.3.1-3)

**To Reproduce**
`./vcpkg install fmilab`

**Failure logs**
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/10039831/config-x64-linux-dbg-err.log)
[config-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/10039832/config-x64-linux-dbg-out.log)
[vcpkg-manifest-install.log](https://github.com/microsoft/vcpkg/files/10039833/vcpkg-manifest-install.log)
[config-x64-linux-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10039834/config-x64-linux-dbg-CMakeCache.txt.log)

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: /home/vsm/vcpkg/downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin/cmake /home/vsm/vcpkg/buildtrees/fmilib/src/ILibrary-2-bb711565e5.clean -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/vsm/vcpkg/packages/fmilib_x64-linux/debug -DFMILIB_BUILD_TESTS=OFF -DFMILIB_BUILD_STATIC_LIB=ON -DFMILIB_BUILD_SHARED_LIB=OFF -DFMILIB_BUILDING_LIBRARY=OFF -DFMILIB_BUILD_WITH_STATIC_RTLIB=OFF -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/vsm/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/vsm/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/vsm/vcpkg -DZ_VCPKG_ROOT_DIR=/home/vsm/vcpkg -D_VCPKG_INSTALLED_DIR=/home/vsm/vsm/vsm-release2/vcpkg_installed -DVCPKG_MANIFEST_INSTALL=OFF -DFETCHCONTENT_FULLY_DISCONNECTED=ON
    Working Directory: /home/vsm/vcpkg/buildtrees/fmilib/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-CMakeCache.txt.log
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-out.log
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:325 (vcpkg_execute_required_process)
  buildtrees/versioning_/versions/fmilib/9bf58761098e936528140d847911dab3613f5983/portfile.cmake:53 (vcpkg_configure_cmake)
  scripts/ports.cmake:147 (include)


error: building fmilib:x64-linux failed with: BUILD_FAILED

```

--- a/ports/fmilib/portfile.cmake
+++ b/ports/fmilib/portfile.cmake
@@ -1,16 +1,11 @@
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
-
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
-vcpkg_download_distfile(ARCHIVE
-    URLS "https://jmodelica.org/fmil/FMILibrary-${VERSION}-src.zip"
-    FILENAME "FMILibrary-${VERSION}-src.zip"
-    SHA512 86e4b5019d8f2a76b01141411845d977fb3949617604de0b34351f23647e3e8b378477de184e1c4f2f59297bc4c7de3155e0edba9099b8924594a36b37b04cc8
-)
-
-vcpkg_extract_source_archive(
-    SOURCE_PATH
-    ARCHIVE "${ARCHIVE}"
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO modelon-community/fmi-library
+    REF "${VERSION}"
+    SHA512 65c2dc11116737e4e2ee91a4ec58d2cf24003774fd6d9b8b1d6521f046be9e8f8a963ebedb50a161ad264927062f41ce757c84563cfe628d47614910e8730349
+    HEAD_REF master
     PATCHES
         0001-remove-install-prefix.patch
         0002-include-sys-stat.h-for-mkdir.patch

--- a/ports/fmilib/vcpkg.json
+++ b/ports/fmilib/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "fmilib",
-  "version": "2.0.3",
-  "port-version": 5,
+  "version": "2.4.1",
   "description": "FMI library is intended as a foundation for applications interfacing FMUs (Functional Mockup Units) that follow FMI Standard. This version of the library supports FMI 1.0 and FMI2.0.",
   "homepage": "https://www.fmi-standard.org/",
   "license": null,

--- a/ports/fmilib/vcpkg.json
+++ b/ports/fmilib/vcpkg.json
@@ -3,7 +3,7 @@
   "version": "2.4.1",
   "description": "FMI library is intended as a foundation for applications interfacing FMUs (Functional Mockup Units) that follow FMI Standard. This version of the library supports FMI 1.0 and FMI2.0.",
   "homepage": "https://www.fmi-standard.org/",
-  "license": null,
+  "license": "BSD-3-Clause",
   "dependencies": [
     {
       "name": "vcpkg-cmake",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2481,8 +2481,8 @@
       "port-version": 0
     },
     "fmilib": {
-      "baseline": "2.0.3",
-      "port-version": 5
+      "baseline": "2.4.1",
+      "port-version": 0
     },
     "fmt": {
       "baseline": "9.1.0",

--- a/versions/f-/fmilib.json
+++ b/versions/f-/fmilib.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "1d4a17469c9429b6e4ab4005e2cffa5a91667498",
+      "git-tree": "34b7f7cc468b8eecb267b5fd750cb1e602d36248",
       "version": "2.4.1",
       "port-version": 0
     },

--- a/versions/f-/fmilib.json
+++ b/versions/f-/fmilib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1d4a17469c9429b6e4ab4005e2cffa5a91667498",
+      "version": "2.4.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "c30d1f2f0827e03141bda90fe3ae2b462f93c116",
       "version": "2.0.3",
       "port-version": 5

